spec/index.bs

```diff
@@ -5170,70 +5187,112 @@ enum GPUStoreOp {
 <dl dfn-type=method dfn-for=GPURenderEncoderBase>
     : <dfn>setPipeline(pipeline)</dfn>
     ::
+        Sets the current {{GPURenderPipeline}}.
 
         <div algorithm="GPURenderEncoderBase.setPipeline">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The render pipeline to use for subsequent drawing commands.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - |pipeline| is [=valid=].
+                        - |pipeline|.{{GPUObjectBase/[[device]]}} is |this|.
+                    </div>
+                1. Set |this|.{{GPURenderEncoderBase/[[pipeline]]}} to be |pipeline|.
+            </div>
         </div>
 
     : <dfn>setIndexBuffer(buffer, indexFormat, offset, size)</dfn>
     ::
+        Sets the current index buffer.
 
         <div algorithm="GPURenderEncoderBase.setIndexBuffer">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setIndexBuffer(buffer, indexFormat, offset, size)">
-                buffer:
-                indexFormat:
-                offset:
-                size:
+                |buffer|: Buffer containing index data to use for subsequent drawing commands.
+                |indexFormat|: Format of the index data contained in |buffer|.
+                |offset|: Offset in bytes into |buffer| where the index data begins.
+                |size|: Size in bytes of the index data in |buffer|.
+                    If `0`, |buffer|.{{GPUBuffer/[[size]]}} - |offset| is used.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setIndexBuffer()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - |buffer| is [=valid=].
+                        - |buffer|.{{GPUObjectBase/[[device]]}} is |this|.
+                        - |offset| &le; |buffer|.{{GPUBuffer/[[size]]}}.
```

Comment:
nit: this condition isn't necessary since we check `offset + size` below. If the goal is to avoid overflows, then the condition below should be `size <= [[size]] - offset`

Reply:
Good point! Thanks!